src/components/Books.js

```diff
@@ -1,21 +1,26 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+// eslint-disable-next-line
+import { getBooksList } from '../redux/Books/books';
 import BookItem from './BookItem';
 
-const Books = ({ bookList }) => (
-  <ul className="books-list-section">
-    {bookList.map((book) => (
-      <BookItem key={book.id} book={book} />
-    ))}
-  </ul>
-);
+const Books = () => {
+  const dispatch = useDispatch();
+  const booksList = useSelector((state) => state.booksReducer);
 
-Books.propTypes = {
-  bookList: PropTypes.arrayOf(PropTypes.shape(
-    {
-      id: PropTypes.string.isRequired,
-    },
-  )).isRequired,
+  useEffect(() => {
+    dispatch(getBooksList());
+  }, [dispatch]);
```

Comment:
[OPTIONAL] every time this component renders this useEffect is dispatching the action against our API. This is something to be aware of, as sometimes we already have the data need it store, and doing unnecessary requests might be expensive. If we check if the books have been already fetched with an if statement we can avoid this behavior.

![bookstore-fetches-everytime](https://user-images.githubusercontent.com/79114233/145660249-1fdb69b4-bb80-451f-a0d7-e1156adce296.gif)